libbeat/monitoring/monitoring.go

```diff
@@ -97,7 +97,16 @@ func SelectConfig(beatCfg BeatConfig) (*common.Config, *report.Settings, error)
 		return monitoringCfg, &report.Settings{Format: report.FormatXPackMonitoringBulk}, nil
 	case beatCfg.Monitoring.Enabled():
 		monitoringCfg := beatCfg.Monitoring
-		return monitoringCfg, &report.Settings{Format: report.FormatBulk}, nil
+
+		var overrideClusterUUID string
+		if monitoringCfg.HasField("override_cluster_uuid") {
+			var err error
+			overrideClusterUUID, err = monitoringCfg.String("override_cluster_uuid", -1)
+			if err != nil {
+				return nil, nil, err
+			}
+		}
```

Comment:
Do not repeat magic strings. There is always the chance of introducing a typo. Better have a `const keyOverrideClusterUUID = "..."` (can be local or global scope). 

btw. go-ucfg has a rich error type: https://godoc.org/github.com/elastic/go-ucfg#Error

e.g.

```
func isMissingKey(err error) {
  if rerr, ok := (err).interface{ Reason() error }; ok {
    return rerr.Reason() == ucfg.ErrMissing
  }
  return true
}
```

then you get:

```
clusterUUID, err := monitoringCfg.String("...", -1)
if err != nil && !isMissingKey(err) {
    return nil, err
}
```

alternative pattern one finds in the code base:

```
var info struct{ ClusterUUID string `config:"..."`}
if err := monitoringCfg.Unpack(&info); err != nil {
    ...
}
```

No strong feelings for one or the other pattern. Just do not type config names as strings multiple times.

Reply:
Fixed in 94d3803f80002884ba8bc5e6490102d85228a9c2.